frontend/dockerfile/parser/parser_test.go

```diff
@@ -175,3 +175,69 @@ func TestParseReturnsScannerErrors(t *testing.T) {
 	_, err := Parse(dockerfile)
 	require.EqualError(t, err, "dockerfile line greater than max allowed size of 65535")
 }
+
+func TestParseExtractsHeredoc(t *testing.T) {
```

Comment:
Some things that may be interesting to test;

Using a `WORD` that "conflicts" with a Dockerfile instruction to verify it's handled correctly;

```dockerfile
FROM busybox
RUN <<COPY
echo hello world
echo foo bar
COPY
```

Comments: I _think_ the expectation would be that comments _within_ a here document are considered part of the here doc content (so not a Dockerfile comment). I'm not sure though if processing currently _first_ strips comments before handling other parsing. Thinking of;

```dockerfile
FROM busybox
# This is a Dockerfile comment
RUN <<EOF
# this is a comment that's part of the RUN (heredoc)
echo hello world
echo foo bar # trailing comments would also be allowed here
EOF
```

Combination of `--options` with heredoc

```dockerfile
FROM busybox
RUN ---mount=type=cache,target=/foo <<EOF
echo hello
EOF
```

Reply:
Currently, the parser should correctly handle both those cases: conflicting `WORD`s and heredoc comments.

However, yup, would be good to nail this behavior down in test cases.